### PR TITLE
[DEVTOOLING-254] Add PCKE Support to .Net Platform SDK

### PR DESCRIPTION
**Pull request by themooks**
Adding Support for PKCE Grant (methods to authenticate and to generate PKCE Code Verifier and to compute Code Challenge)

### CODE REVIEW

**Review by dginty4 — APPROVED** [review]

**Review by carnellj-genesys — APPROVED** [review]

File: resources/sdk/pureclouddotnet/templates/README.mustache [hunk 1]
```diff
@@ -123,11 +123,39 @@ using {{packageName}}.Extensions;
...
+Then call the _PostTokenPKCE_ extension method of _ApiClient_ with your orgName and encodedSamlAssertion
+
+```csharp
+var accessTokenInfo = Configuration.Default.ApiClient.PostTokenPKCE(clientId, redirectUri, codeVerifier, authCode);
+Console.WriteLine("Access token=" + accessTokenInfo.AccessToken);
+```
+
+The SDK provides methods to generate a PKCE Code Verifier and to compute PKCE Code Challenge.
+
+```csharp
```
Comment (HemanthDogiparthi12) [inline resources/sdk/pureclouddotnet/templates/README.mustache:154]: may be we can keep this section on top of the ReadMe for API call. ?

File: resources/sdk/pureclouddotnet/extensions/Extensions/AuthExtensions.cs [hunk 2]
```diff
@@ -178,10 +180,136 @@ public static AuthTokenInfo PostTokenSaml2Bearer(this ApiClient apiClient, strin
...
+            return randomString;
+        }
+
+        /// <summary>
+        /// Compute Base64Url PKCE Code Challenge from Code Verifier.
+        /// </summary>
+        /// <param name="apiClient"></param>
+        /// <param name="code"></param>
+        /// <returns></returns>
+        public static string ComputePKCECodeChallenge(this ApiClient apiClient, string code)
```
[the remark this answers is not in the document]
Comment (themooks) [reply resources/sdk/pureclouddotnet/extensions/Extensions/AuthExtensions.cs:225]: Explained in the Readme.

File: resources/sdk/pureclouddotnet/templates/README.mustache (hunk 1 above)
Comment (themooks) [reply resources/sdk/pureclouddotnet/templates/README.mustache:154]: I understand. But these 2 methods are just to help people generating a PKCE Code Verifier or to compure a PKCE Code Challenge. They can always follow the RFC and build their own method.
The AuthorizePKCEGrant is the important one.
But I plan to add quick-hits/PKCE guide and sample codes early next year.

**Comment by dginty4** [comment]
@HemanthDogiparthi12 Am I good to merge this?

**Pull request merged**